### DISCUSSION TITLE
How are the response times saved?

### DISCUSSION THREAD

**Discussion by inesmtc** [jspsych/jsPsych#943] [Q&A] [answered]
Hello everyone,

I've launched my experiment on Pavlovia and recently downloaded the results. I was surprised to see that all the RTs did't have the same precision depending on the subject. Some were integer and others where decimal numbers. Is there an explication? That's not a big deal but I would like to know the cause of this. 
Moreover, I put a trial duration to 1500ms and noticed one RT equals to 1501.21ms. Is that due to the calculation time of the computer/Pavlovia's server or something like this?

Thanks by advance!

**Comment by jodeleeuw** [comment, marked as answer]
Hi @inesmtc 

*Why don't all subjects have the same rt precision?*

For most plugins jspsych uses the function `performance.now()` to measure the clock. Different browsers may implement the return value of this function differently, with varying degrees of precision. In any case, even if the clock measurements are precise to microsecond levels the RT measurement certainly isn't that precise due to other sources of error in the measurement pipeline.

*Why did you get an rt longer than the trial duration?*

It's possible that:
* The trial duration was slightly longer than requested. JavaScript clocks are difficult to control to this degree of accuracy so a few milliseconds of error is to be expected occasionally.
* The response was recorded just before the trial ended and the response timing was measured just after the trial was terminated by the timeout.

**Reply by inesmtc** [reply]
Hi @jodeleeuw ,

Thanks for your answer :D
